external/fv3fit/fv3fit/data/base.py

```diff
@@ -0,0 +1,46 @@
+import abc
+import tensorflow as tf
+from typing import Optional, Sequence
+import dacite
+
+
+class TFDatasetLoader(abc.ABC):
+    @abc.abstractmethod
+    def get_data(
+        self, local_download_path: Optional[str], variable_names: Sequence[str],
+    ) -> tf.data.Dataset:
+        """
+        Args:
+            local_download_path: if provided, cache data locally at this path
+            variable_names: names of variables to include when loading data
+        Returns:
+            dataset containing requested variables
+        """
+        ...
+
+    @classmethod
+    def from_dict(cls, kwargs) -> "TFDatasetLoader":
```

Comment:
Ya, we use this pattern as well in `loaders`. I'm not sure if this is the better of the two between this or registration, kept this pattern here for minimal changes.

I want to avoid inheritance other than for abstract classes/protocols, but I may be able to turn this into a helper function in `vcm`.

Reply:
The logic got removed in switching to a registry pattern.